bpf/bpf_overlay.c

```diff
@@ -353,7 +356,7 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
 		}
 #endif
 		/* See comment at equivalent code in handle_ipv6() */
-		if (info && identity_is_remote_node(*identity))
+		if (info && (identity_is_remote_node(*identity) || identity_is_world_ipv4(*identity)))
```

Comment:
With SNAT + tunneling, packets from LB SNATed with a remote node IP can be wrongly resolved as remote node identity?

Reply:
Agree, this is a bit too invasive right now I think and would conflict with SNATed LB connections.

Maybe just pass a `is_dsr` flag up from `nodeport_lb*()`? And use that in `from-overlay` to decide whether to resolve the identity from ipcache.

Reply:
I see.
How about using `#ifdef`? 
I think we can use it like `#ifdef ENABLE_DSR` when DSR is enabled.

Reply:
With `#ifdef`, in-cluster(pod to pod) communication over tunnel relies on ipcache too instead of the tunnel header when DSR is enabled, right? It would still be invasive, perhaps.

Reply:
Oh that makes sense. Yes, we should decide whether to use ipcache per packet not config.

Reply:
I pushed some changes adding `is_dsr` flag :)